dev-dotnet/dotnet-sdk/dotnet-sdk-5.0.203.ebuild

```
@@ -0,0 +1,238 @@
+# (the --base-tarball option might be worth using now it's added)
+
+# https://github.com/dotnet/runtime/blob/release/5.0/src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json#L27
+LICENSE="MIT Apache-2.0 BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+system-libunwind system-bootstrap"
+
+RDEPEND="
+	app-crypt/mit-krb5
+	dev-libs/icu
+	dev-util/lttng-ust
+	system-libunwind? ( sys-libs/libunwind )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-util/cmake
+	dev-vcs/git
+	dev-dotnet/dotnet-reference-packages
+	system-bootstrap? ( >=dev-dotnet/dotnet-sdk-${BOOT_PV} )
+"
+
+CHECKREQS_DISK_BUILD="20G"
+CHECKREQS_DISK_USR="1200M"
+
+# Use ninja unless emake is explicitly requested
+# Adapted from cmake.eclass
+case ${CMAKE_MAKEFILE_GENERATOR:-ninja} in
+	emake)
+		BDEPEND+=" sys-devel/make"
+		CMAKE_GENERATOR_FLAG=""
+		;;
+	ninja|*)
+		BDEPEND+=" dev-util/ninja"
+		CMAKE_GENERATOR_FLAG="-ninja"
+		;;
+esac
+
+# If the package can't be built using gcc currently, set this to "true"
+DOTNET_FORCE_CLANG="true"
+
+if ${DOTNET_FORCE_CLANG}; then
+	BDEPEND+=" sys-devel/clang"
+fi
+
+dotnet_unpack() {
+	if ! use system-bootstrap; then
+		mkdir "${WORKDIR}"/dotnet || die
+		cd "${WORKDIR}"/dotnet || die
+		unpack "dotnet-sdk-${BOOT_PV}-linux-x64.tar.gz"
+
+		mkdir source-artifacts || die
+		cd source-artifacts || die
+		unpack "Private.SourceBuilt.Artifacts.${BOOT_ARTIFACTS_PV}.tar.gz"
+	fi
+}
+
+dotnet_find() {
+	if ! use system-bootstrap; then
+		DOTNET_ROOT="${WORKDIR}/dotnet"
+		return
+	fi
+
+	for x in "${DOTNET_ROOT}" "${BROOT}"/usr/lib/dotnet; do
+		if [[ -d "${x}" ]] && [[ -d "${x}/source-artifacts" ]]; then
+			DOTNET_ROOT="${x}"
+			break
+		fi
+	done
+
+	if [[ ! -d "${DOTNET_ROOT}" ]]; then
+		die "Can't find installed .NET SDK (including Source-built Artifacts)"
+	fi
+}
+
+dotnet_find_reference() {
+	# Handle the case where we're bootstrapping with a binary sdk,
+	#  but the reference-packages have just been installed in /usr/lib
+	for x in "${DOTNET_ROOT}"/reference-packages "${BROOT}"/usr/lib/dotnet/reference-packages; do
+		if [[ -d "${x}" ]]; then
+			DOTNET_REFERENCE="${x}"
+			break
+		fi
+	done
+
+	if [[ ! -d "${DOTNET_REFERENCE}" ]]; then
+		die "Can't find installed .NET Source-build Reference Packages"
+	fi
+}
+
+src_unpack() {
+	unpack "${MY_P}.tar.gz"
+	dotnet_unpack
+}
+
+src_prepare() {
+	# add_patch <submodule> <file.patch>
+	add_patch() {
+		cp "${2}" "patches/${1}/9000-$(basename "${2}")" || die
+	}
+
+	eapply "${FILESDIR}"/dotnet-sdk-5.0.203-additional-arguments.patch
+	add_patch sdk "${DISTDIR}"/dotnet-sdk-telemetry-optout.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-add-gentoo-ids.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-additional-arguments.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-as-needed.patch
+	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-locate-ranlib.patch
+
```

```suggestion
	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-locate-ranlib.patch
	add_patch runtime "${FILESDIR}"/dotnet-sdk-5.0.203-runtime-lld-build.patch
```

Please add the following to fix building the runtime with LLD:

```
--- a/eng/native/configurecompiler.cmake
+++ b/eng/native/configurecompiler.cmake
@@ -595,7 +595,7 @@
         ERROR_QUIET
         OUTPUT_VARIABLE ldVersionOutput)
 
-    if("${ldVersionOutput}" MATCHES "GNU ld" OR "${ldVersionOutput}" MATCHES "GNU gold")
+    if("${ldVersionOutput}" MATCHES "GNU ld" OR "${ldVersionOutput}" MATCHES "GNU gold" OR "${ldVersionOutput}" MATCHES "GNU linkers")
         set(LD_GNU 1)
     elseif("${ldVersionOutput}" MATCHES "Solaris Link")
         set(LD_SOLARIS 1)

```

---

I see this is only required for clang built with `default-lld`, annoyingly enough. Added.